typings.d.ts

```diff
@@ -0,0 +1,80 @@
+declare module 'gavel' {
+  export enum RESTMethods {
+    GET = 'GET',
+    POST = 'POST',
+    PUT = 'PUT',
+    PATCH = 'PATCH',
+    DELETE = 'DELETE',
+    OPTIONS = 'OPTIONS'
+  }
+
+  export interface HTTPMessage {
+    method: RESTMethods;
+    uri: string;
+    statusCode: string | number;
+    headers: Record<string, string> | string;
+    body: Record<string, any> | string;
+    /**
+     * [JSON Schema](https://json-schema.org/) Draft V3-4.
+     */
+    bodySchema: Record<string, any> | string;
```

Comment:
It would be neat if `Record<string, any>` would be type definition for JSON Schema (Draft X) itself.

Reply:
Let me see if there is any typings exposed by JSONSchema in any way, so we could inherit it here. That'd be great.

Reply:
It turns out there is `@types/json-schema`, which provides type definitions for JSONSchema Draft V4, V6, and V7. We can include the V4 in Gavel type definitions.

Reply:
Although `@types/json-schema` provides type definition for JSON Schema Draft V4, since we don't have the same for V3, we would need to use a union type with `Record`, making any more precise typings useless. TypeScript won't validate the validity of JSON Schema V4 since, if invalid, it satisfies the `Record` type.